notifications/src/main/java/dk/shape/games/notifications/presentation/NotificationSettingsFragment.kt

```diff
@@ -81,10 +81,16 @@ class NotificationSettingsFragment : Fragment() {
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        action.eventIds?.let { eventIds ->
-            fetchNotifications(eventIds)
-        } ?: action.getEventIds { eventIds ->
-            fetchNotifications(eventIds)
+        when (action) {
+            is NotificationSettingsAction.IncludeAllEvents ->
+                fetchNotifications(
+                    (action as NotificationSettingsAction.IncludeAllEvents).eventIds
```

Comment:
```suggestion
                    action.eventIds
```
the action is already cast at this point

Reply:
it doesn't allow to smart cast it, since it has "open or custom getter"